com.civicrm.hrjobroles/js/controllers/HRJobRolesController.js

```diff
@@ -330,10 +328,36 @@ define(['controllers/controllers'], function(controllers){
             $scope.checkNewRole = function checkNewRole() {
 
                 return (typeof $scope.edit_data['new_role_id'] === 'undefined'
-                    || typeof $scope.edit_data['new_role_id']['title'] === 'undefined'
-                    || $scope.edit_data['new_role_id']['title'] == ''
-                    || typeof $scope.edit_data['new_role_id']['job_contract_id'] === 'undefined'
-                    || $scope.edit_data['new_role_id']['job_contract_id'] == '');
+                || typeof $scope.edit_data['new_role_id']['title'] === 'undefined'
+                || $scope.edit_data['new_role_id']['title'] == ''
+                || typeof $scope.edit_data['new_role_id']['job_contract_id'] === 'undefined'
+                || $scope.edit_data['new_role_id']['job_contract_id'] == '');
+            };
+
+            /**
+             * @description Parse dates so they can be correctly read by server.
+             * Use od DateFactory.CreateDate() takes care of timezone issue.
+             * @param date {string|timestamp|Date}
+             * @returns {Date}
+             */
+            $scope.parseDate = function (date) {
```

Comment:
DateFactory uses moment.js to do that. 
All I wanted to do was to get offset before converting Date object to moment.js object.
In the process of convertion this information is somehow lost, so methods utcOffset(), zone() return 0 (or -0) instead of correct value. Not sure why is that, that's why i decided that getting offset before convertion is more secure.